contracts/factory/src/contract.rs

```diff
@@ -69,24 +86,31 @@ impl FactoryTrait for Factory {
             &lp_init_info.stake_init_info,
         );
 
+        let lp_wasm_hash = get_config(&env).lp_wasm_hash;
+        let stake_wasm_hash = get_config(&env).stake_wasm_hash;
+        let token_wasm_hash = get_config(&env).token_wasm_hash;
+
         let lp_contract_address = deploy_lp_contract(
             &env,
-            lp_init_info.lp_wasm_hash,
+            lp_wasm_hash,
             &lp_init_info.token_init_info.token_a,
             &lp_init_info.token_init_info.token_b,
         );
 
         let init_fn: Symbol = Symbol::new(&env, "initialize");
         let init_fn_args: Vec<Val> = (
-            lp_init_info.admin,
-            lp_init_info.share_token_decimals,
-            lp_init_info.swap_fee_bps,
-            lp_init_info.fee_recipient,
-            lp_init_info.max_allowed_slippage_bps,
-            lp_init_info.max_allowed_spread_bps,
-            lp_init_info.max_referral_bps,
-            lp_init_info.token_init_info.clone(),
-            lp_init_info.stake_init_info,
+            // lp_init_info.admin,
+            // lp_init_info.share_token_decimals,
+            // lp_init_info.swap_fee_bps,
+            // lp_init_info.fee_recipient,
+            // lp_init_info.max_allowed_slippage_bps,
+            // lp_init_info.max_allowed_spread_bps,
+            // lp_init_info.max_referral_bps,
+            // lp_init_info.token_init_info.clone(),
+            // lp_init_info.stake_init_info,
```

Comment:
Should it be removed? What happend with those variables?

Reply:
We are adding two extra fields to `factory` - `stake_wasm_hash` and `token_wasm_hash`. We also pass down these to `pool` and `pool_stable` and this jumps above the limit of 10 arguments per method. We cannot override this even with `clippy`.

That's why, instead of sending too many arguments, I pass only lp_init_info and we `unzip` it's content inside `pool` and `stable_pool`

Reply:
commented out code remove, closing this